addons/l10n_it_edi/models/res_partner.py

```diff
@@ -31,21 +31,135 @@ class ResPartner(models.Model):
             "Destination Code must have between 6 and 7 characters."),
     ]
 
-    @api.model
-    def _l10n_it_normalize_codice_fiscale(self, codice):
-        if codice and re.match(r'^IT[0-9]{11}$', codice):
-            return codice[2:13]
-        return codice
+    def _l10n_it_edi_get_values(self):
+        """ Generates all partner values needed by l10n_it_edi XML export """
+        if not self:
+            return False
+
+        self.ensure_one()
+        return {
+            'self_id': self,
+            'is_company': self.is_company,
+            'country_code': self._l10n_it_edi_normalized_country(),
+            'state_code': self._l10n_it_edi_normalize_state(),
+            'zip': self._l10n_it_edi_normalize_zip(),
+            'vat': self._l10n_it_edi_normalized_vat(),
+            'codice_fiscale': self._l10n_it_edi_normalized_codice_fiscale(),
+            'in_eu': self._l10n_it_edi_in_eu(),
+            'pa_index': self._l10n_it_edi_normalized_pa_index(),
+            'phone': self.phone,
+            'mobile': self.mobile,
+            'email': self.email,
+            'pec_email': self.l10n_it_pec_email,
+            'display_name': self.display_name,
+            'first_name': ' '.join(self.name.split()[:1]),
+            'last_name': ' '.join(self.name.split()[1:]),
+            'street': self.street,
+            'street2': self.street2,
+            'city': self.city,
```

Comment:
I can not find an easy way immediately myself to avoid copying these values, but there should be one.

Reply:
Changed.